zookeeper-server/src/main/java/org/apache/zookeeper/common/ClientX509Util.java

```diff
@@ -99,6 +113,14 @@ public SslContext createNettySslContextForClient(ZKConfig config)
         }
     }
 
+    private void logTcnativeOcsp(boolean enable) {
+        if (enable && !OpenSsl.isOcspSupported()) {
+            // SslContextBuilder.enableOcsp() doesn't do anything, unless the default BoringSSL
+            // tcnative dependency is replaced with an OpenSsl one.
+            LOG.warn("Trying to enable OCSP for tcnative OpenSSL provider, but it is not supported. The setting will be ignored");
```

Comment:
"Trying to enable OCSP stapling, but the selected SSL provider {name} doesn't support it"

Reply:
There is no API to get the tcnative implementation name.

The closest thing I could find is OpenSsl.versionString() .

Reply:
Would you like to add that to the log line ?

Reply:
What does `sslProvider.toString()` give?